cpp/server/ct-server.cc

```diff
@@ -282,6 +289,31 @@ int main(int argc, char* argv[]) {
   ClusterStateController<LoggedCertificate> cluster_controller(
       &internal_pool, &consistent_store, &election_);
 
+  // Ensure we can only be the master if we've got sufficient local replication
+  util::SyncTask sth_watch_task(&internal_pool);
+  consistent_store.WatchServingSTH(
+      [&election_, &cluster_controller](const Update<SignedTreeHead>& sth) {
+        if (!sth.exists_) {
+          LOG(WARNING) << "Cluster has no Serving STH - leaving election.";
+          election_.StopElection();
```

Comment:
I don't see anything restarting the participation in the election once we get a serving STH or we catch up with the replication? Missing bit, or will be in a later pull request?

It's okay if it's for a later pull request, but I'd put a TODO until then.